head.html

```diff
@@ -20,6 +20,90 @@
 
   [miloStyles, miloUtils].forEach((tag) => tag.setAttribute('rel', 'preload'));
   document.head.append(miloStyles, miloUtils);
+
+  function buildLink(language, url) {
+    const link = document.createElement('link');
+    link.setAttribute('rel', 'alternate');
+    link.setAttribute('hreflang', language);
+    link.setAttribute('href', url);
+    return link;
+  }
+
+  const userAgentMeta = document.querySelector('meta[name="hreflinksuseragents"]');
+  const allowedAgents = userAgentMeta && userAgentMeta.content.split(',');
+  const userAgentString = window.navigator.userAgent;
+  const isAllowedAgent = allowedAgents && allowedAgents.some((agent) => userAgentString.includes(agent.trim()));
+
+  const LOCALES = ['jp', 'kr', 'de', 'fr', 'au', 'uk'];
+
+  async function fetchAndParseSitemap() {
+    const sitemapOrigin = 'https://business.adobe.com';
+    const { origin, pathname } = window.location;
+
+    let sitemapPath = 'sitemap.xml';
+    const localeMatch = LOCALES.find(locale => pathname.startsWith(`/${locale}/`));
+
+    if (localeMatch) {
+      sitemapPath = `sitemap-${locale}.xml`;
+    }
+    
+    try {
+      const response = await fetch(`${origin}/blog/${sitemapPath}`);
+      if (!response.ok) {
+        console.warn('Failed to fetch sitemap:', response.status);
+        return;
+      }
+      
+      const xmlText = await response.text();
+      const parser = new DOMParser();
+      const xmlDoc = parser.parseFromString(xmlText, 'text/xml');
+      const parseError = xmlDoc.querySelector('parsererror');
+      if (parseError) {
+        return;
+      }
+      
+      const urlElements = xmlDoc.querySelectorAll('url');
+      let currentUrlElement = null;
+      const currentPageUrl = pathname !== '/' ? `${sitemapOrigin}${pathname}` : `${sitemapOrigin}/`;
+
+      for (const urlElement of urlElements) {
+        const loc = urlElement.querySelector('loc')?.textContent;
+        if (loc === currentPageUrl || loc === currentPageUrl.replace(/\/$/, '')) {
+          currentUrlElement = urlElement;
+          break;
+        }
+      }
+      
+      if (!currentUrlElement) {
+        console.warn('Current page not found in sitemap');
+        return;
+      }
+      
+      const alternateLinks = currentUrlElement.querySelectorAll('link[rel="alternate"]');
+      const linkElements = [];
+
+      alternateLinks.forEach(altLink => {
+        const hreflang = altLink.getAttribute('hreflang');
+        const href = altLink.getAttribute('href');
+        
+        if (hreflang && href) {
+          linkElements.push(buildLink(hreflang, href));
+        }
+      });
+      
+       const titleElement = document.head.querySelector('title');
+      if (linkElements.length > 0 && titleElement) {
+        titleElement.after(...linkElements);
+      }
+    } catch (error) {
+      console.error('Error fetching or parsing sitemap:', error);
+    }
+  }
+  
+  if (isAllowedAgent) {
+    fetchAndParseSitemap();
+  }
```

Comment:
Should this be in a closure or embed isAllowedAgent inside of fetchAndParseSitemap? Otherwise fetchAndParseSitemap and all the consts are in a global scope. Could be an IIFE.

Reply:
Good catch, it is the same with da-bacom. I created an issue for that. https://github.com/adobecom/da-bacom/issues/85